sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitions.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+
+import org.apache.spark.annotation.Experimental;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.analysis.NoSuchPartitionException;
+import org.apache.spark.sql.catalyst.analysis.PartitionAlreadyExistsException;
+
+/**
+ * A partition interface of {@link Table}.
+ * A partition is composed of identifier and properties,
+ * and properties contains metadata information of the partition.
+ * <p>
+ * These APIs are used to modify table partition identifier or partition metadata,
+ * in some cases, they will change actual value of table data as well.
+ *
+ * @since 3.0.0
+ */
+@Experimental
+public interface SupportsPartitions extends Table {
+
+    /**
+     * Create a partition in table.
+     *
+     * @param ident a new partition identifier
+     * @param properties the metadata of a partition
+     * @throws PartitionAlreadyExistsException If a partition already exists for the identifier
+     */
+    void createPartition(
+            InternalRow ident,
+            Map<String, String> properties) throws PartitionAlreadyExistsException;
+
+    /**
+     * Drop a partition from table.
+     *
+     * @param ident a partition identifier
+     * @return true if a partition was deleted, false if no partition exists for the identifier
+     */
+    Boolean dropPartition(InternalRow ident);
+
+    /**
+     * Rename a Partition from old identifier to new identifier with no metadata changed.
+     *
+     * @param oldIdent the partition identifier of the existing partition
+     * @param newIdent the new partition identifier of the partition
+     * @throws NoSuchPartitionException If the partition identifier to rename doesn't exist
+     * @throws PartitionAlreadyExistsException If the new partition identifier already exists
+     */
+    void renamePartition(
+            InternalRow oldIdent,
+            InternalRow newIdent) throws NoSuchPartitionException, PartitionAlreadyExistsException;
+
+    /**
+     * Replace the partition metadata of the existing partition.
+     *
+     * @param ident the partition identifier of the existing partition
+     * @param properties the new metadata of the partition
+     * @throws NoSuchPartitionException If the partition identifier to rename doesn't exist
+     */
+    void replacePartitionMetadata(
+            InternalRow ident,
+            Map<String, String> properties) throws NoSuchPartitionException;
+
+    /**
+     * Retrieve the partition metadata of the existing partition.
+     *
+     * @param ident a partition identifier
+     * @return the metadata of the partition
+     */
+    Map<String, String> getPartitionMetadata(InternalRow ident);
```

Comment:
`get` is a misleading verb in Java APIs because it typically means that a value is fetched cheaply from a local variable. It's better to use a verb like `load` if the operation is likely to be backed by a network call.

Reply:
good, it has been changed.

Reply:
Thanks for updating it. This also fits with the other APIs, which use `load`.